test/functional/shada/shada_spec.lua

```diff
@@ -250,6 +250,21 @@ describe('ShaDa support code', function()
     end
   end)
 
+  it(
+    'allows writing and reading ShaDa files when shadafile=NONE but supplied with a filename',
+    function()
+      nvim_command('set shadafile=NONE')
+      eq(0, exc_exec('wshada ' .. shada_fname))
+      eq(1, read_shada_file(shada_fname)[1].type)
+
+      wshada('Some text file')
+      eq(
+        'Vim(rshada):E576: Error while reading ShaDa file: last entry specified that it occupies 109 bytes, but file ended earlier',
+        exc_exec('rshada ' .. shada_fname)
```

Comment:
untested, sample code:
```suggestion
      eq(
        'Vim(rshada):E576: Error while reading ShaDa file: last entry specified that it occupies 109 bytes, but file ended earlier',
        t.pcall_err(n.command, 'rshada ' .. shada_fname)
```